pac4j-core/src/main/java/org/pac4j/core/engine/DefaultSecurityLogic.java

```diff
@@ -202,19 +218,19 @@ protected void saveRequestedUrl(final C context, final List<Client> currentClien
     /**
      * Perform a redirection to start the login process of the first indirect client.
      *
-     * @param context the web context
+     * @param context        the web context
      * @param currentClients the current clients
      * @return the performed redirection
      */
     protected HttpAction redirectToIdentityProvider(final C context, final List<Client> currentClients) {
         final IndirectClient currentClient = (IndirectClient) currentClients.get(0);
-        return currentClient.redirect(context);
+        return (HttpAction) currentClient.redirect(context).get();
```

Comment:
.get() alone…

Reply:
How should we deal with no HttpAction returned? I've thrown an Error by now.

Reply:
I don't think it makes sense to have an optional `HttpAction`  here. The signature of the `redirect` method should not have an `Optional`.
As we are working on the API, I'm wondering if this `redirect` should belong to the `IndirectClient`  class instead of the `Client` interface.